### ISSUE TITLE
Auto-rotate based on user location - stopped working in Safari

### ISSUE THREAD

**Issue by afitzgerrell** [GCRC/nunaliit#316]
Earlier in the week when I was testing auto-detection of my location (and orientating the atlas appropriately) in the ArcticCBM prototype (http://devel.gcrc.carleton.ca:8098/index.html) with Safari (7.1.4), the atlas was correctly, automatically displaying EPSG:3573 for me, whether opening the atlas anew or switching between the main module and the Tiles or Clusters menu options.  Now, however, it is defaulting to and sticking to EPSG:3571 (although I can manually switch the projection with the widget or base layers).

I have reset Safari so that I could re-allow GCRC to use my location, but that doesn't seem to do anything for me and the atlas is still defaulting to EPSG:3571.  I don't know for certain that this is an atlas issue rather than a Safari issue where it's not passing along my location properly.  I'm not sure how to verify that it should be pinned on one or the other, although Firefox and Chrome are maintaining my location default seamlessly so far.  

If you have questions or suggestions of something to try, please let me know.  Tomorrow (Friday 3/27) is a holiday for me, but I'll be back in the office on Monday.

**Comment by jpfiset** [comment]
Are you still having the same issue with Safari? Do you know anyone else facing the same problem?

There is a line reporting the current position in the logs. Are you seeing it? From my logs, it looks like this:

getCurrentPosition() {coords: {accuracy: 38, altitude: null, altitudeAccuracy: null, heading: null, latitude: 45.3532532, longitude: -75.7241903, speed: null}, timestamp: 1428504078355}

**Comment by afitzgerrell** [comment]
Hi JP - Safari is still behaving poorly in terms of this issue.  I have asked a coworker to try opening the dev atlas in Safari and it acted similarly for him - he allowed sharing of his location and yet, the map didn't auto-rotate to EPSG:3573 as we'd expect it to.  

I'm not familiar with where I should look to find the logs - do you mean Safari-specific logs, from extensions like developer tools or firebug (that I have for Chrome and Firefox)?

**Comment by jpfiset** [comment]
Yes, I would need to see the Javascript logs, as with the ones offered by Firebug.

**Comment by afitzgerrell** [comment]
I added web inspector and now see an error is being thrown regarding getCurrentPosition (when I refresh the page):

![screen shot 2015-04-09 at 2 31 37 pm](https://cloud.githubusercontent.com/assets/8062694/7076236/2e88cf80-dec5-11e4-8e6d-a66fa85857f9.png)
expanding the Position error shows:
![screen shot 2015-04-09 at 2 32 35 pm](https://cloud.githubusercontent.com/assets/8062694/7076259/55c117c4-dec5-11e4-8aa7-e979aadf4ed2.png)
Let me know if you'd prefer for me to copy/past text from the log here rather than screen grabs.

**Comment by jpfiset** [comment]
The reported error is POSITION_UNAVAILABLE. This means that the browser is failing from obtaining the location. There is little recourse Nunaliit can take if the browser refuses to return the location.

Look at this post: http://stackoverflow.com/questions/17574919/why-is-geolocate-failing-in-safari-desktop

Does it corresponds to your experience?

**Comment by afitzgerrell** [comment]
Hi JP - sorry for the long time it has taken me to respond.  I read the stackoverflow article - it corresponds quite well to my experience.  I just connected via Wi-Fi to the test atlas and it was able to determine my location and rotate appropriately.  If I turn Wi-Fi off and start a new Safari session, it goes back to not being able to recognize my location and leaves the atlas at the default EPSG:3571.

**Comment by ahayes** [comment]
JP, is that the only error sent for all reasons for position not being available? (user declined, browser can't find itself, not supported, etc?)
